### PR TITLE
load posts and like functionality

### PR DESCRIPTION
**Pull request by sashaboi**
Added Feed from live posts. 
Added like / dislike functionality .

Still to add : 
1.icon should change on liked 
2.loader to fix glitchy vibe of the first load. 
3.fix css of layout it feels broken for some reason.

### CODE REVIEW

File: src/components/PostCard/PostCard.jsx [hunk 1]
```diff
@@ -2,38 +2,66 @@ import React from 'react';
...
+    const token = localStorage.getItem('token');
+    if (token === null) {
+      navigate('/auth/login');
+    }
+    const header = { headers: { authorization: token } };
+    if (hasliked) {
+      console.log('dislike');
+      axios.post(`api/posts/dislike/${post._id}`, {}, header).then(
+        response => {
+          console.log('dislike response', response.data.posts);
```
Comment (yashpatel1797) [inline src/components/PostCard/PostCard.jsx:26]: remove console log

File: src/components/PostCard/PostCard.jsx [hunk 2]
```diff
@@ -2,38 +2,66 @@ import React from 'react';
...
+  const LikePostHandler = () => {
+    const hasliked = post.likes.likedBy.some(obj => obj._id === userObj._id);
+    const token = localStorage.getItem('token');
+    if (token === null) {
+      navigate('/auth/login');
+    }
+    const header = { headers: { authorization: token } };
+    if (hasliked) {
+      console.log('dislike');
+      axios.post(`api/posts/dislike/${post._id}`, {}, header).then(
```
Comment (yashpatel1797) [inline src/components/PostCard/PostCard.jsx:24]: just a suggestion you can use `async/await` here.

File: src/context/User-context.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+const UserProvider = ({ children }) => {
+  const [userObj, setUserObj] = useState();
+  return (
+    <UserContext.Provider value={{ userObj, setUserObj }}>
+      {children}
+    </UserContext.Provider>
+  );
+};
+const UseUser = () => useContext(UserContext);
```
Comment (yashpatel1797) [inline src/context/User-context.js]: for custom hooks follow this practice `useUser`

**Review by yashpatel1797 — APPROVED** [review]
- Folder structure looks good
- code is clean
- design looks good.

File: src/components/PostCard/PostCard.jsx (hunk 2 above)
Comment (sashaboi) [reply src/components/PostCard/PostCard.jsx:24]: noted.

**Pull request merged**